lib/Repository.js

```diff
@@ -187,8 +187,10 @@ class Repository extends Requestable {
     * @return {Promise} - the promise for the http request
     */
    listCommits(options, cb) {
-      options = options || {};
-
+      if (typeof options === 'function') {
```

Comment:
It looks like the options object is not considered optional according to the JS docs. I think it's also worth thinking about existing users, so we don't change the interface for them going forwards without a major semantic version increase.
I think if you include both the default value setting of `options = options || {};` as well as the callback code, it should work for both situations. Also update the Jsdoc comments to mark options as optional.

Reply:
I added the original functionality back along with the new code. The options parameter is already marked as optional in the [JSDocs](http://github-tools.github.io/github/docs/3.1.0/Repository.html#listCommits).